src/lib/metaphysicsV2.coffee

```diff
@@ -0,0 +1,60 @@
+Q = require 'bluebird-q'
+qs = require 'qs'
+request = require 'superagent'
+{ extend, some } = require 'underscore'
+{ METAPHYSICS_ENDPOINT, API_REQUEST_TIMEOUT, REQUEST_ID } = require('sharify').data
+ip = require 'ip'
+
+resolveIPv4 = (ipAddress) ->
+  if ip.isV6Format(ipAddress)? and ipAddress.indexOf('::ffff') >= 0
+    return ipAddress.split('::ffff:')[1]
+  return ipAddress
+
+resolveProxies = (req) ->
+  ipAddress = resolveIPv4(req.connection.remoteAddress)
+  if req?.headers?["x-forwarded-for"]?
+    return req.headers["x-forwarded-for"] + ", " + ipAddress
+  else
+    return ipAddress
+
+metaphysics = ({ query, variables, req } = {}) ->
```

Comment:
Can this be updated to reflect the `v2` function? `metaphysicsV2`

Reply:
This is file local so it's not that important to be very explicit but let's do that anyway :+1: